core/trino-main/src/main/java/io/trino/operator/function/TableFunctionOperator.java

```diff
@@ -285,21 +294,36 @@ public TableFunctionOperator(
         PagesIndex pagesIndex = pagesIndexFactory.newPagesIndex(sourceTypes, expectedPositions);
         HashStrategies hashStrategies = new HashStrategies(pagesIndex, partitionChannels, prePartitionedChannels, sortChannels, sortOrders, preSortedPrefix);
 
-        this.outputPages = pageBuffer.pages()
-                .transform(new PartitionAndSort(pagesIndex, hashStrategies, processEmptyInput))
-                .flatMap(groupPagesIndex -> pagesIndexToTableFunctionPartitions(
-                        groupPagesIndex,
-                        hashStrategies,
-                        tableFunctionProvider,
-                        session,
-                        functionHandle,
-                        properChannelsCount,
-                        passThroughSourcesCount,
-                        requiredChannels,
-                        markerChannels,
-                        passThroughSpecifications,
-                        processEmptyInput))
-                .flatMap(TableFunctionPartition::toOutputPages);
+        if (preferStreaming && passThroughSpecifications.isEmpty() && sortOrders.isEmpty()) {
+            StreamingWorkProcessor streamingWorkProcessor = new StreamingWorkProcessor(processEmptyInput, getOnlyElement(requiredChannels));
+            this.outputPages = pageBuffer.pages()
+                    .transform(streamingWorkProcessor.toTableFunctionInput())
+                    .transform(new TableFunctionWorkProcessor(tableFunctionProvider.getDataProcessor(session, functionHandle)))
+                    .map(page -> {
+                        Block[] blocks = new Block[properChannelsCount];
+                        for (int i = 0; i < properChannelsCount; i++) {
+                            blocks[i] = page.getBlock(i);
+                        }
+                        return new Page(blocks);
```

Comment:
What Blocks do we filter out here? The pass-through markers? We shouldn't create an inconsistent state where `passThroughSpecifications.isEmpty()` and yet there are pass-through markers appended.

Reply:
I have removed this in https://github.com/trinodb/trino/pull/25046